src/CaptureClient/CaptureClient.cpp

```diff
@@ -265,16 +262,16 @@ bool CaptureClient::StopCapture() {
   {
     absl::MutexLock lock(&state_mutex_);
     if (state_ == State::kStarting) {
-      // Block and wait until the state is not kStarting
-      bool (*is_not_starting)(State*) = [](State* state) { return *state != State::kStarting; };
-      state_mutex_.Await(absl::Condition(is_not_starting, &state_));
+      LOG("StopCapture ignored, because it is starting and cannot be stopped at this stage.");
```

Comment:
From a user perspective, this means, they have to press the button again to stop the capture, right? 

What's the reason for not keeping the wait here?

Reply:
it leads to a deadlock. OnCaptureStarted waits on something to be executed on the main thread while main thread waits for OnCaptureStarted to complete.

Reply:
Got it, thanks for clarifying!